### PR TITLE
Fix for WFCORE-1594

### PR DESCRIPTION
**Pull request by jfdenise**
Support for streams in the CLI. Warning, some unit tests are based on Incremental/exploded deployment, so mut be merged once the exploded deployments is in.

### CODE REVIEW

**Comment by jfdenise** [comment]
retest this please!

**Comment by marekkopecky** [comment]
Tests from this PR depends on https://github.com/wildfly/wildfly-core/pull/1675 . So before this PR can be merged, https://github.com/wildfly/wildfly-core/pull/1675 should be merged

**Comment by bstansberry** [comment]
retest this please

**Comment by marekkopecky** [comment]
retest this please

**Comment by marekkopecky** [comment]
retest this please

**Comment by jfdenise** [comment]
retest this please

File: cli/src/main/java/org/jboss/as/cli/Attachements.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ *
+ * @author jdenise@redhat.com
+ */
+public final class Attachements {
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/Attachements.java]: This looks like French spelling? English doesn't have the initial 'e'.

**Comment by jfdenise** [comment]
@bstansberry , Attachement vs Attachment, yes typical French... I Fixed all occurrences of it (found a lot off them...). Thank-you.

**Comment by jfdenise** [comment]
retest this please

**Comment by jfdenise** [comment]
retest this please

File: cli/src/main/java/org/jboss/as/cli/Attachments.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        this.immutable = immutable;
+    }
+
+    public Attachments() {
+        this(false);
+    }
+
+    public int addFileAttachment(String path) {
+        if (immutable) {
+            throw new UnsupportedOperationException("Attachments can't be muted");
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/Attachments.java]: s/muted/mutated

**Comment by bstansberry** [comment]
Trivial comment above aside, this looks good. But it needs a rebase now, probably because I merged the 'timeout' PR.

**Comment by jfdenise** [comment]
retest this please

**Comment by jfdenise** [comment]
retest this please

**Pull request merged**